dissect/target/loaders/mqtt.py

```diff
@@ -423,13 +426,147 @@ def connect(self) -> None:
+    """Checks if the value is a valid IP address."""
+    try:
+        ipaddress.ip_address(value)
+        return True
+    except ValueError:
+        return False
+
+
+def host_name_or_ip_address(value: str) -> str:
+    """
+    Validate if the given value is a valid hostname or IP address.
+
+    This function is intended to be used as a type for argparse arguments.
+
+    Args:
+        value (str): The value to be validated as a hostname or IP address.
+    Returns:
+        str: The validated hostname or IP address.
+    Raises:
+        argparse.ArgumentTypeError: If the value is not a valid hostname or IP address.
+    """
+
+    if host_name(value) or ip_address(value):
+        return value
+
+    raise argparse.ArgumentTypeError(f"Invalid hostname or IP address specified: '{value}'")
+
+
+def port(value: str) -> int:
+    """
+    Convert a string value to an integer representing a valid port number.
+
+    This function is intended to be used as a type for argparse arguments.
+
+    Args:
+        value (str): The string representation of the port number.
+    Returns:
+        int: The port number as an integer.
+    Raises:
+        argparse.ArgumentTypeError: If the port number is not an integer or out of the valid range (1-65535).
+    """
+
+    try:
+        port = int(value)
+        if port < 1 or port > 65535:
+            raise argparse.ArgumentTypeError("Port number must be between 1 and 65535.")
+        return port
+    except ValueError:
+        raise argparse.ArgumentTypeError(f"Invalid port number specified: '{value}'")
+
+
+def case(value: str) -> str:
+    """
+    Validates that the given value is a valid case name consisting of
+    alphanumeric characters and underscores only.
+
+    This function is intended to be used as a type for argparse arguments.
+
+    Args:
+        value (str): The case name to validate.
+
+    Returns:
+        str: The validated case name if it matches the required pattern.
+
+    Raises:
+        argparse.ArgumentTypeError: If the case name does not match the required pattern.
+    """
+
+    if re.match(r"^[a-zA-Z0-9_]+$", value):
+        return value
+
+    raise argparse.ArgumentTypeError(f"Invalid case name specified: '{value}'")
+
+
+def file_exists(file_path: str) -> str:
+    """Checks if the file exists."""
+    if not os.path.isfile(file_path):
+        raise argparse.ArgumentTypeError(f"File does not exist: '{file_path}'")
+    return file_path
+
+
+@arg("--mqtt-peers", type=strictly_positive, dest="peers", help="minimum number of peers to await for first alias")
+@arg(
+    "--mqtt-case",
+    type=case,
+    dest="case",
+    help="case name (broker will determine if you are allowed to access this data)",
+)
+@arg("--mqtt-port", type=port, dest="port", help="broker connection port")
+@arg("--mqtt-broker", type=host_name_or_ip_address, dest="broker", help="broker ip-address")
+@arg("--mqtt-key", type=file_exists, dest="key", help="private key file")
```

Comment:
For reviewer: In other files these typically have `type=path`.

The advantage of file_exists is that you we get an exception immediately, and not in a deep stack trace where the error might be more difficult to comprehend. 

Let me know what you think.